### PR TITLE
Add contrib docs checklist for adding new acl token

### PR DESCRIPTION
**Pull request by lkysow**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -497,6 +497,88 @@ rebase the branch on main, fixing any conflicts along the way before the code ca
 
 ---
 
+## Adding a new ACL Token
+
+Checklist for getting server-acl-init to generate a new ACL token assuming the new token
```
Comment (t-eckert) [inline CONTRIBUTING.md:502]:
```suggestion
This is a checklist for getting server-acl-init to generate a new ACL token assuming the new token
```

**Review by thisisnotashwin — APPROVED** [review]
Thanks for doing this!

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -497,6 +497,88 @@ rebase the branch on main, fixing any conflicts along the way before the code ca
...
+is named `foo`.
```
Comment (t-eckert) [inline CONTRIBUTING.md:503]:
```suggestion
This is a checklist for getting `server-acl-init` to generate a new ACL token named `foo`.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -497,6 +497,88 @@ rebase the branch on main, fixing any conflicts along the way before the code ca
...
+
+### Control Plane
+
+* `control-plane/subcommand/server-acl-init/command.go`
+    * Add `flagCreateFooToken bool` to vars list
+    * Initialize flag in `init`
```
Comment (t-eckert) [inline CONTRIBUTING.md:509]:
```suggestion
    * [ ] Add `flagCreateFooToken bool` to vars list
    * [ ] Initialize flag in `init`
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -497,6 +497,88 @@ rebase the branch on main, fixing any conflicts along the way before the code ca
...
+
+      ```go
+      c.flags.BoolVar(&c.flagCreateFooToken, "create-foo-token", false,
+        "<docs for flag>")
+      ```
+    * Add `if` statement in `Run` to create your token (follow placement of other tokens).
```
Comment (t-eckert) [inline CONTRIBUTING.md:515]:
```suggestion
    * [ ] Add `if` statement in `Run` to create your token (follow placement of other tokens).
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -497,6 +497,88 @@ rebase the branch on main, fixing any conflicts along the way before the code ca
...
+      ```go
+      if c.flagCreateFooToken {
+          err := c.createLocalACL("foo", fooRules, consulDC, isPrimary, consulClient)
+          if err != nil {
+              c.log.Error(err.Error())
+              return 1
+          }
+      }
+      ```
+* `control-plane/subcommand/server-acl-init/rules.go`
```
Comment (t-eckert) [inline CONTRIBUTING.md:527]:
```suggestion
* [ ] `control-plane/subcommand/server-acl-init/rules.go`
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -497,6 +497,88 @@ rebase the branch on main, fixing any conflicts along the way before the code ca
...
+    * Add a function that outputs your rules using a template
+      (if the rules are always the same us a `const`):
+      ```go
+      func (c *Command) fooRules() (string, error) {
+      ```
+* `control-plane/subcommand/server-acl-init/rules_test.go`
```
Comment (t-eckert) [inline CONTRIBUTING.md:533]:
```suggestion
* [ ] `control-plane/subcommand/server-acl-init/rules_test.go`
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -497,6 +497,88 @@ rebase the branch on main, fixing any conflicts along the way before the code ca
...
+* `control-plane/subcommand/server-acl-init/command.go`
+    * Add `flagCreateFooToken bool` to vars list
+    * Initialize flag in `init`
+
+      ```go
+      c.flags.BoolVar(&c.flagCreateFooToken, "create-foo-token", false,
+        "<docs for flag>")
+      ```
+    * Add `if` statement in `Run` to create your token (follow placement of other tokens).
+      You'll need to decide if you need a local token (`createLocalACL`) or global token `createGlobalACL`.
```
Comment (t-eckert) [inline CONTRIBUTING.md:516]:
```suggestion
      You'll need to decide if you need a local token, `createLocalACL`, or global token, `createGlobalACL`.
```

Consistency in formatting.

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -497,6 +497,88 @@ rebase the branch on main, fixing any conflicts along the way before the code ca
...
+          err := c.createLocalACL("foo", fooRules, consulDC, isPrimary, consulClient)
+          if err != nil {
+              c.log.Error(err.Error())
+              return 1
+          }
+      }
+      ```
+* `control-plane/subcommand/server-acl-init/rules.go`
+    * Add a function that outputs your rules using a template
+      (if the rules are always the same us a `const`):
```
Comment (t-eckert) [inline CONTRIBUTING.md:529]:
```suggestion
      (if the rules do not change, use a `const`):
```

"same" should be used when it's explicit what it is the same as. This is implicitly the same as it was in the past so it's easier to just say it doesn't change.

File: CONTRIBUTING.md (hunk 3 above)
Comment (t-eckert) [reply CONTRIBUTING.md:509]: I would do this throughout for the sub-items because it will make a nice checklist the user can click to tick off items.

**Review by t-eckert — APPROVED** [review]
Looks great! I just had a few notes.

File: CONTRIBUTING.md (hunk 3 above)
Comment (lkysow) [reply CONTRIBUTING.md:509]: I thought about this but how can they check off the items in the contributing.md though?

Comment (t-eckert) [reply CONTRIBUTING.md:509]: I think it just sets a toggle in `localStorage`. Does it work on your branch if you add it?

Comment (lkysow) [reply CONTRIBUTING.md]: nope 😢 : https://github.com/hashicorp/consul-k8s/blob/ce5916890b8a280e86c9914fae54b9b743c20bc0/CONTRIBUTING.md

**Pull request merged**